### PR TITLE
[metasrv] refine: load configs from toml file

### LINKED ISSUE #3088: add metasrv startup config file: `--config-file <fn>` or `-c <fn>`
Load config from a toml file instead of from command line args if `--config-file` or `-c` is provided.

Related file:
https://github.com/datafuselabs/databend/blob/4cdb600ec4ee6d6ee9da0fb3c1de23cafebf62dd/metasrv/src/configs/config.rs#L53

`main()` needs to load config from specified file:
https://github.com/datafuselabs/databend/blob/4cdb600ec4ee6d6ee9da0fb3c1de23cafebf62dd/metasrv/src/bin/metasrv.rs#L33

Add the corresponding test in `metasrv/tests/it/config/`.

Reference:

https://github.com/datafuselabs/databend/blob/99596d2b066d9472455d51d5ce8c9b533b2d8aea/query/src/configs/config.rs#L58-L60

https://github.com/datafuselabs/databend/blob/4cdb600ec4ee6d6ee9da0fb3c1de23cafebf62dd/query/src/bin/databend-query.rs#L36-L45

### PR DESCRIPTION
**Pull request by liubaohai**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Changelog

- Improvement

## Related Issues

Fixes #3088

### CODE REVIEW

**Comment by BohuTANG** [comment]
Hi,
It is useful to execute `make lint` locally :D

File: metasrv/src/configs/config.rs [hunk 1]
```diff
@@ -113,4 +135,80 @@ impl Config {
     pub fn tls_rpc_server_enabled(&self) -> bool {
         !self.flight_tls_server_key.is_empty() && !self.flight_tls_server_cert.is_empty()
     }
+
+    /// First load configs from args.
+    /// If config file is not empty, e.g: `-c xx.toml`, reload config from the file.
+    /// Prefer to use environment variables in cloud native deployment.
+    /// Override configs base on environment variables.
+    pub fn load() -> Result<Self, ErrorCode> {
```
Comment (drmingdrmer) [inline metasrv/src/configs/config.rs:143]: Nice comment!

File: metasrv/src/configs/config.rs [hunk 2]
```diff
@@ -13,11 +13,25 @@
...
+    ($struct:expr, $field:tt, $field_type: ty, $env:expr) => {
+        if let Some(env_var) = std::env::var_os($env) {
+            $struct.$field = env_var
+                .into_string()
+                .expect(format!("connot convert {} to string", $env).as_str())
+                .parse::<$field_type>()
+                .expect(format!("cannot convert {} to {}", $env, stringify!($field_type)).as_str());
+        }
+    };
+}
```
Comment (drmingdrmer) [inline metasrv/src/configs/config.rs:33]: What about 
```
    ($field:tt, $field_type: ty, $env:expr) => {
   // ... 
   $field = ...
```

It seems `$struct` is always used together with`$field` and can be reduced.

**Review by drmingdrmer — COMMENTED** [review]
Nice shot!

Click the `ready for review` when you get `make lint` happily passed locally and on CI. :DDD

**Comment by liubaohai** [comment]
> 

Thanks for reminding

**Comment by BohuTANG** [comment]
/lgtm

Looks great, thank you @liubaohai !

**Comment by liubaohai** [comment]
> @liubaohai Looks like your github account is not same with your commits id, it will not show in [Databend contributors wall,](https://github.com/datafuselabs/databend/graphs/contributors) we would like you were there :D

Ok, I will change my git user account.

**Pull request merged**